test/new-e2e/tests/agent-platform/package-signing/package_signing_test.go

```diff
@@ -6,50 +6,61 @@ package packagesigningtest
 
 import (
 	"encoding/json"
+	"flag"
 	"fmt"
 	"os"
+	"regexp"
 
+	"github.com/DataDog/datadog-agent/comp/metadata/packagesigning/packagesigningimpl"
 	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
 	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
 	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/params"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2params"
 	"github.com/stretchr/testify/assert"
 
 	"testing"
 )
 
-type Payload struct {
-	Hostname  string           `json:"hostname"`
-	Timestamp int64            `json:"timestamp"`
-	Metadata  *signingMetadata `json:"signing_metadata"`
-}
+var osName = flag.String("osname", "", "os to test, eg debian or redhat")
 
-type signingMetadata struct {
-	SigningKeys []SigningKey `json:"signing_keys"`
+type packageSigningTestSuite struct {
+	e2e.Suite[e2e.AgentEnv]
+	osName string
 }
 
-type SigningKey struct {
-	Fingerprint    string         `json:"signing_key_fingerprint"`
-	ExpirationDate string         `json:"signing_key_expiration_date"`
-	KeyType        string         `json:"signing_key_type"`
-	Repositories   []repositories `json:"repositories"`
-}
+func TestPackageSigningComponent(t *testing.T) {
 
-type repositories struct {
-	RepoName string `json:"repo_name"`
-}
+	e2eOSMapping := map[string]ec2os.Type{
+		"debian":      ec2os.DebianOS,
+		"ubuntu":      ec2os.UbuntuOS,
+		"centos":      ec2os.CentOS,
+		"amazonlinux": ec2os.AmazonLinuxOS,
+		"redhat":      ec2os.RedHatOS,
+		"rhel":        ec2os.RedHatOS,
+		"sles":        ec2os.SuseOS,
+		"fedora":      ec2os.FedoraOS,
+		"suse":        ec2os.SuseOS,
+		"rocky":       ec2os.RockyLinux,
+	}
 
-type packageSigningTestSuite struct {
-	e2e.Suite[e2e.FakeIntakeEnv]
-}
+	var testedOS ec2os.Type
+	nonAlpha := regexp.MustCompile("[^a-zA-Z]")
+	rawOS := nonAlpha.ReplaceAllString(*osName, "")
+	if value, ok := e2eOSMapping[rawOS]; ok {
+		testedOS = value
+	} else {
+		t.Fatalf("OS %s not supported", *osName)
+	}
 
-func TestPackageSigningComponent(t *testing.T) {
-	t.Run(fmt.Sprintln("Test package signing on default ubuntu"), func(tt *testing.T) {
-		// tt.Parallel()
-		fmt.Println("Testing on ubuntu")
-		e2e.Run[e2e.FakeIntakeEnv](tt,
-			&packageSigningTestSuite{},
-			e2e.FakeIntakeStackDefWithDefaultVMAndAgentClient(),
-			params.WithStackName(fmt.Sprintf("pkgSigning-ubuntu-%s", os.Getenv("CI_PIPELINE_ID"))),
+	t.Run(fmt.Sprintf("Test package signing on %s\n", rawOS), func(tt *testing.T) {
+		tt.Parallel()
+		fmt.Printf("Testing on %s\n", rawOS)
```

Comment:
It seems redundant to have this line printed since the test name has the `raw` as part of it.

Reply:
Indeed it was a copy/paste, I can remove this